tasks/Task.cpp

```diff
@@ -13,6 +14,55 @@ Task::~Task()
 {
 }
 
+PWMTable loadCommandToPWMTable(std::string const& csv_file_path)
+{
+
+    PWMTable table;
+
+    std::ifstream csv_file(csv_file_path);
+
+    if (!csv_file.is_open()) {
+        throw std::runtime_error("could not open csv file");
+    }
+
+    std::string row;
+    while (std::getline(csv_file, row)) {
+        std::size_t comma_i = row.find(',');
+        table.cmd.push_back(std::atof(row.substr(0, comma_i).c_str()));
+        table.duty_cycle_width.push_back(std::atoi(row.substr(comma_i + 1).c_str()));
+    }
+
+    return table;
+}
+
+uint32_t Task::computePWMCommand(float command) const
+{
+    if (command == 0) {
+        return m_no_actuation_pwm_command;
+    }
+
+    const auto& lut = m_cmd_to_pwm_lut;
+    if (command <= lut.cmd.front()) { // saturation
+        return lut.duty_cycle_width.front();
+    }
+
+    if (command >= lut.cmd.back()) { // saturation
+        return lut.duty_cycle_width.back();
+    }
+
+    // PWMTable is assumed to be small, that is a couple hundred lines
+    std::size_t i = 1; // i is the upper bound limit
+    for (; command > lut.cmd[i]; i++)
+        ;
```

Comment:
I could use for the cleaner argument but I don't think
```
auto upper = std::upper_bound(lut.cmd.begin(), lut.cmd.end(), command);
std::size_t i = std::distance(lut.cmd.begin(), upper);
```
is better than a plain for loop regarding readability.

> even if the marginal efficiency gain is minimal as PWMTable is small

I don't want to get in the efficiency rabbit hole here, because it is hardly ever what really matters. But, given that pwm table is small a linear search would take advantage of cache memory whereas upper_bound (a binary search) would throw this advantage away and make much more access to main memory and, probably be indeed slower